### PR TITLE
 [WordFilter] Handle no configured filters

### LINKED ISSUE #454: [WordFilter] Handle no configured filters
Let's handle the case where we haven't added filters yet a little more gracefully.
```
Ignoring exception in on_message
Traceback (most recent call last):
  File "/home/ubuntu/github/renchan/lib/python3.8/site-packages/discord.py-1.7.3-py3.8.egg/discord/client.py", line 343, in _run_event
    await coro(*args, **kwargs)
  File "/home/ubuntu/github/renchan/Ren/cogs/highlight/highlight.py", line 765, in onMessage
    await self.checkHighlights(msg)
  File "/home/ubuntu/github/renchan/Ren/cogs/highlight/highlight.py", line 640, in checkHighlights
    elif await self.wordFilter.containsFilterableWords(msg):
  File "/home/ubuntu/github/renchan/Ren/cogs/wordfilter/wordfilter.py", line 445, in containsFilterableWords
    filteredMsg = _filterWord(filters, filteredMsg)
  File "/home/ubuntu/github/renchan/Ren/cogs/wordfilter/wordfilter.py", line 565, in _filterWord
    regex = reFormat.format(*words)
IndexError: Replacement index 0 out of range for positional args tuple
Ignoring exception in on_message
Traceback (most recent call last):
  File "/home/ubuntu/github/renchan/lib/python3.8/site-packages/discord.py-1.7.3-py3.8.egg/discord/client.py", line 343, in _run_event
    await coro(*args, **kwargs)
  File "/home/ubuntu/github/renchan/Ren/cogs/wordfilter/wordfilter.py", line 549, in on_message
    await self.checkWords(msg)
  File "/home/ubuntu/github/renchan/Ren/cogs/wordfilter/wordfilter.py", line 494, in checkWords
    filteredMsg = _filterWord(filteredWords, filteredMsg)
  File "/home/ubuntu/github/renchan/Ren/cogs/wordfilter/wordfilter.py", line 565, in _filterWord
    regex = reFormat.format(*words)
IndexError: Replacement index 0 out of range for positional args tuple

```

To work around this for now, add one filter with the `[p]wf re add` command

### PR DESCRIPTION
**Pull request by jlacsamana**
This PR fixes #454:
- reworked _wordfilter() to return message unmodified if filter list is empty

### CODE REVIEW

File: cogs/wordfilter/constants.py [hunk 1]
```diff
@@ -7,4 +7,9 @@
 KEY_FILTERS = "filters"
 KEY_CMD_DENIED = "commandDenied"
 KEY_TOGGLE_MOD = "toggleMod"
-BASE = {KEY_CHANNEL_IDS: [], KEY_FILTERS: [], KEY_CMD_DENIED: [], KEY_TOGGLE_MOD: False}
+BASE = {
+    KEY_CHANNEL_IDS: [],
+    KEY_FILTERS: [],
+    KEY_CMD_DENIED: [],
+    KEY_TOGGLE_MOD: False,
+}
```
Comment (Injabie3) [inline cogs/wordfilter/constants.py]: nit: remove these changes from this PR.

File: cogs/wordfilter/wordfilter.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 To filter words in a more smart/useful way than simply detecting and
 deleting a message.
 """
+from os import stat
```
Comment (Injabie3) [inline cogs/wordfilter/wordfilter.py]: Not needed.

File: cogs/wordfilter/wordfilter.py [hunk 3]
```diff
@@ -86,6 +87,7 @@ async def addFilter(self, ctx, word: str):
                 "`Word Filter:` `{0}` was added to the filter in the "
                 "guild **{1}**".format(word, guildName)
             )
+
```
Comment (Injabie3) [inline cogs/wordfilter/wordfilter.py]: nit: remove whitespace change.

File: cogs/wordfilter/wordfilter.py [hunk 4]
```diff
@@ -440,6 +442,7 @@ async def containsFilterableWords(self, msg):
 
         filteredMsg = msg.content
         filters = await self.config.guild(msg.guild).get_attr(KEY_FILTERS)()
+
```
Comment (Injabie3) [inline cogs/wordfilter/wordfilter.py]: nit: remove whitespace change.

File: cogs/wordfilter/wordfilter.py [hunk 5]
```diff
@@ -490,6 +493,7 @@ async def checkWords(
 
         try:
             filteredMsg = _filterWord(filteredWords, filteredMsg)
+
```
Comment (Injabie3) [inline cogs/wordfilter/wordfilter.py]: nit: remove whitespace change.

File: cogs/wordfilter/wordfilter.py [hunk 6]
```diff
@@ -557,13 +561,22 @@ def _censorMatch(matchobj):
...
+    else:
+        # Combine all filter words into one regex
+        numFilters = numWords - 1
+        reFormat = r"\b(?:" + (r"{}|") * numFilters + r"{})\b"
+        regex = reFormat.format(*words)
+
+        # sees how many times each filtered word appears in string and tallies them up in the config file
+        for word in words:
+            wordInstances = len(re.findall(regex, word))
+
```
Comment (Injabie3) [inline cogs/wordfilter/wordfilter.py]: Can we remove this change since it's not related to this issue?

**Review by Injabie3 — CHANGES_REQUESTED** [review]
Thanks, can we remove the stuff that isn't related to the fix?

File: cogs/wordfilter/wordfilter.py [hunk 7]
```diff
@@ -557,13 +557,15 @@ def _censorMatch(matchobj):
 
 
 def _filterWord(words, string):
-    # Combine all filter words into one regex
-    numFilters = len(words) - 1
-    reFormat = r"\b(?:" + (r"{}|") * numFilters + r"{})\b"
-    regex = reFormat.format(*words)
-
-    # Replace the offending string with the correct number of stars.
-    return re.sub(regex, _censorMatch, string, flags=re.IGNORECASE)
```
Comment (Injabie3) [inline cogs/wordfilter/wordfilter.py:566]: What happened to this return?

**Review by Injabie3 — CHANGES_REQUESTED** [review]
One last thing

File: cogs/wordfilter/wordfilter.py (hunk 7 above)
Comment (jlacsamana) [reply cogs/wordfilter/wordfilter.py:566]: accidental delete when i was manually splitting the PRs lmaoo

**Review by Injabie3 — APPROVED** [review]
Thanks, LGTM

**Pull request merged**